plugins/woocommerce/src/Admin/API/MarketingCampaigns.php

```diff
@@ -132,6 +132,69 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Get formatted price based on Price type.
+	 *
+	 * This uses plugins/woocommerce/i18n/currency-info.php and plugins/woocommerce/i18n/locale-info.php to get option object based on $price->currency.
+	 *
+	 * Example:
+	 *
+	 * - When $price->currency is 'USD' and $price->value is '1000', it should return '$1000.00'.
+	 * - When $price->currency is 'JPY' and $price->value is '1000', it should return '¥1,000'.
+	 * - When $price->currency is 'AED' and $price->value is '1000', it should return '5.000,00 د.إ'.
+	 *
+	 * @param Price $price Price object.
+	 * @return String formatted price.
+	 */
+	private function get_formatted_price( $price ) {
+		$currency_info_all = include WC()->plugin_path() . '/i18n/currency-info.php';
+		$currency_info     = $currency_info_all[ $price->get_currency() ]['default'];
+
+		$locale_info_all = include WC()->plugin_path() . '/i18n/locale-info.php';
+		$locale_index    = array_search( $price->get_currency(), array_column( $locale_info_all, 'currency_code' ), true );
+		$num_decimals    = array_values( $locale_info_all )[ $locale_index ]['num_decimals'];
+
+		add_filter(
```

Comment:
The inline filter should be removed after the operation is complete. Otherwise it can affect other parts of the code.

Reply:
Maybe you could use `price_format` parameter in `wc_price` instead of the filter?

Reply:
Something like

```PHP
$currencyFormats = [
    'right'       => '%2$s%1$s',
    'left_space'  => '%1$s&nbsp;%2$s',
    'right_space' => '%2$s&nbsp;%1$s',
];

$price_format = isset($currencyFormats[$currency_pos]) ? $currencyFormats[$currency_pos] : '%1$s%2$s';
```

Reply:
Thanks! I was thinking using filter is good (for some reason I was thinking it was a good "reuse"), but on hindsight yours is better and simpler! Done in https://github.com/woocommerce/woocommerce/pull/44917/commits/c9091c8bf8871e270c6b20df6da916bd69131e38 with a bit of minor change from yours.